File: StackLinkedList.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DataStructures
{
    class LinkedListStack
    {
        public class Node
        {
            public Node next;
            public Object data;
            public Node(Object data)
            {
                this.data = data;
            }
        }
        public Node top;
        public int size;
        public LinkedListStack()
        {
            size = 0;
        }

        public void push(Object data)
        {
            Node newNode = new Node(data);
            newNode.next = top;
            top = newNode;
            size++;
        }

        public Object pop()
        {
            if (isEmpty())
            {
                return null;
            }
            Object deletedNode = top.data;
            top = top.next;
            size--;
            return deletedNode;
        }
        public Object peek()
        {
            if (isEmpty())
            {
                return null;
            }
            return top.data;
        }
        public Boolean isEmpty()
        {
            if (size == 0)
            {
                Console.WriteLine("List is empty!");
                return true;
            }
            return false;
        }
        public int Size()
        {
            return size;
        }
        public void printStack()
        {
            if (isEmpty())
            {
                return;
            }
            Node current = top;
            while (current != null)
            {
                Console.Write("{0} ", current.data);
                current = current.next;
            }
            Console.WriteLine();
        }
    }
}

File: Java/Stacks/StackLinkedList.java
/*
 * Implementation of a stack using a Linked List
 * A stack is LIFO
 */
public class StackLinkedList {
	public Node top; //head in normal linked list
	public int size = 0;
	
	public void push(int data){
		Node pushNode = new Node(data);
		pushNode.next = top; //set next node to current top
		top = pushNode; //set new top to the new node
		size++; //increment size of list
	}
	public void pop(){
		if(top == null){
			return;
		}
		top = top.next; //pop the top node
		size--;
	}
	public int top(){ //returns data in top node
		return top.data;
	}
	
	public boolean isEmpty(){ //checks if list is empty
		if(top == null){
			return true;
		}
		return false;
	}
}
